C++ Algorithms/lcsa.cpp

```diff
@@ -0,0 +1,87 @@
+// Longest Common Subsequence Implementation
+#include<bits/stdc++.h>
+
+using namespace std;
+
+void printlcs(int n, char *p, string A, int i, int j){
+	//Function to print the LCS
+	if(i==0 || j==0){
+		return;
+	}
+	if(*((p+i*n) + j) == '\\'){
+		printlcs(n, p, A, i-1, j-1);
+		cout<<A[i-1];
+	}
+	else if(*((p+i*n) + j) == '|'){
+		printlcs(n, p, A, i-1, j);
+	}
+	else if(*((p+i*n) + j) == '-'){
+		printlcs(n, p, A, i, j-1);
+	}
+}
+
+void lcs(string A, string B, int m, int n){
+	//Function to calculate longest common subsequence
+	int l[m+1][n+1], i, j;
+	char p[m+1][n+1];
```

Comment:
recheck these two lines. final code should be working fine
tip: m and n should be constant or you can use dynamic array

Reply:
you can use this example:

        int arr[10][10];
	int x = 10;
	int** l = new int *[x];
	l[m + 1][n + 1];
	int i, j;
	char** p = new char *[x];
	p[m + 1][n + 1];

Reply:
The code is working fine as variables m and n get their values via function arguments. So they are in a sense constants during array declaration.
![Output](https://user-images.githubusercontent.com/47164154/66709104-b29df480-ed7a-11e9-9ca8-a1763952cfec.png)